src/overlays/OrdersOverlay.tsx

```diff
@@ -0,0 +1,70 @@
+import { Socket } from 'socket.io-client';
+import { useCallback, useEffect, useState } from 'react';
+import { getAllOrders, Order } from '../api';
+import OrderList from '../components/OrderList';
+
+interface ShowOrdersEvent {
+  orders: Order[];
+}
+
+interface Props {
+  socket: Socket;
+}
+
+export default function OrdersOverlay({ socket }: Props) {
+  const [orders, setOrders] = useState<Order[]>([]);
+
+  useEffect(() => {
+    getAllOrders()
+      .then((ordersRes) => {
+        if (!ordersRes.data) return;
+        setOrders(ordersRes.data);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+
+    socket.on('orders', (newOrderSet) => {
+      setOrders((newOrderSet as ShowOrdersEvent[])[0].orders);
+    });
+
+    return () => {
+      socket.removeAllListeners();
```

Comment:
Even though I remember we use this in different places as well, I don't think this is good behavior. Right now, we might also delete listeners from handlers, which then seize to function. I think it is better to declare the callback function a separate arrow function and then remove this specific callback on unmount.

Reply:
Should be fixed!